packages/core/src/util/mergeSchemaUI.ts

```diff
@@ -0,0 +1,74 @@
+import type { PartialSchema } from "../types/index.js"
+
+type UISchemaAttributes<TPartialSchema extends PartialSchema> = {
+  [attributeName in keyof TPartialSchema["attributes"]]?: {
+    control: {
+      ui: TPartialSchema["attributes"][attributeName]["control"]["ui"]
+    }
+  }
+}
+
+interface PartialUISchema<TPartialSchema extends PartialSchema> {
+  ui?: PartialSchema["ui"]
+  attributes?: UISchemaAttributes<TPartialSchema>
+}
+
+export type PartialUISchemas<
+  TPartialSchemas extends Record<string, PartialSchema>,
+> = {
+  [schemaName in keyof TPartialSchemas]?: PartialUISchema<
+    TPartialSchemas[schemaName]
+  >
+}
+
+/**
+ * Merges a set of partial schemas with a set of partial UI schemas. If the partial UI schema
+ * or any of it's attibutes do not exist in the set of partial schemas, they are ignored.
+ */
+export function mergeSchemaUI<
```

Comment:
This isn't being used anywhere; should it be?

Reply:
We should have a simple test for it at least.

Reply:
there's tests for it - it's not used internally anywhere, just exported for consumers of hatchify